pmd-core/src/main/java/net/sourceforge/pmd/cpd/XMLRenderer.java

```diff
@@ -101,10 +107,31 @@ public String render(Iterator<Match> matches) {
 
     @Override
     public void render(Iterator<Match> matches, Writer writer) throws IOException {
-        Document doc = createDocument();
-        Element root = doc.createElement("pmd-cpd");
+        render(new CPDReport(matches, Collections.<String, Integer>emptyMap()), writer);
+    }
+
+    @Override
+    public void render(final CPDReport report, final Writer writer) throws IOException {
+        final Document doc = createDocument();
+        final Element root = doc.createElement("pmd-cpd");
+        final Map<String, Integer> numberOfTokensPerFile = report.getNumberOfTokensPerFile();
+        final Iterator<Match> matches = report.getMatches();
         doc.appendChild(root);
 
+        final List<Map.Entry<String, Integer>> entries = new ArrayList<>(numberOfTokensPerFile.entrySet());
+        Collections.sort(entries, new Comparator<Map.Entry<String, Integer>>() {
+            @Override
+            public int compare(final Map.Entry<String, Integer> entry1, final Map.Entry<String, Integer> entry2) {
+                return entry1.getKey().compareTo(entry2.getKey());
+            }
+        });
```

Comment:
We could move this sorting into CPDReport - when we specify, that the CPDReports returns a SortedMap and the report returns the files pre-sorted. Not sure, if this is actually useful or whether using SortedMap would be too restrictive for CPDReport.